swagger_to/swagger.py

```diff
@@ -289,22 +306,34 @@ def parse_path(adict: MutableMapping[str, Any]) -> Tuple[Path, List[str]]:
     return pth, errors
 
 
-def parse_yaml(stream: Any) -> Tuple[Swagger, List[str]]:
+def parse_yaml(stream: Any, with_line_number: bool) -> Tuple[Swagger, List[str]]:
     """
     Parses the Swagger specification from the given text.
 
     :param stream: YAML representation of the Swagger spec satisfying file interface
+    :param with_line_number: if set, the swagger definition will contain line numbers at each node
     :return: Swagger specification, list of errors
     """
     # adapted from https://stackoverflow.com/questions/5121931/in-python-how-can-you-load-yaml-mappings-as-ordereddicts
+    # and https://stackoverflow.com/questions/13319067/parsing-yaml-return-with-line-number
+
     object_pairs_hook = collections.OrderedDict
 
     class OrderedLoader(yaml.SafeLoader):
-        pass
-
-    def construct_mapping(loader, node):
+        def compose_node(self, parent, index):
+            # the line number where the previous token has ended (plus empty lines)
+            node = Composer.compose_node(self, parent, index)
+            if with_line_number:
+                line = self.line
+                node.__line__ = line + 1
+            return node
+
+    def construct_mapping(loader, node, deep=False):
         loader.flatten_mapping(node)
-        return object_pairs_hook(loader.construct_pairs(node))
+        mapping = Constructor.construct_pairs(loader, node, deep=deep)
+        if with_line_number:
+            mapping.append(('__line__', node.__line__))
```

Comment:
(Comment A) Let's rename it to `__lineno__` to conform with `inspect` module (see, for example, https://stackoverflow.com/questions/6810999/how-to-determine-file-function-and-line-number)